zebra/zapi_msg.c

```diff
@@ -3144,12 +3144,29 @@ void zserv_handle_commands(struct zserv *client, struct stream_fifo *fifo)
 			goto continue_loop;
 		}
 
-		zapi_parse_header(msg, &hdr);
+		if (STREAM_READABLE(msg) < ZEBRA_HEADER_SIZE
+		    || !zapi_parse_header(msg, &hdr)) {
+			if (IS_ZEBRA_DEBUG_PACKET && IS_ZEBRA_DEBUG_RECV)
+				zlog_debug(
+					"ZAPI message is %zu bytes long but the minimum packet size is %u; dropping",
+					STREAM_READABLE(msg),
+					ZEBRA_HEADER_SIZE);
+			goto continue_loop;
+		}
 
 		if (IS_ZEBRA_DEBUG_PACKET && IS_ZEBRA_DEBUG_RECV
 		    && IS_ZEBRA_DEBUG_DETAIL)
 			zserv_log_message(NULL, msg, &hdr);
 
+		if (hdr.length < ZEBRA_HEADER_SIZE) {
```

Comment:
This code for the stream is tested in zserv_read, right?  Do we need to test this one again?

Reply:
Oh, yeah...actually this whole commit isn't needed, silly me :)